### PR TITLE
Add AODP

### PR DESCRIPTION
**Pull request by peterk87**
This PR adds a recipe for [AODP](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2363-3) (v2.5.0.1), a bioinformatics tool "that generates oligonucleotide signatures for sequences in the FASTA format and for groups in a phylogeny in the Newick tree format".

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/aodp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - gxx_linux-64 ==7.3.0 # [linux]
```
Comment (bgruening) [inline recipes/aodp/meta.yaml:19]: Can you please remove this compiler and also perl/bioperl from the build section.

File: recipes/aodp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    - perl-bioperl >=1.7.2
+
+test:
+  commands:
+    - aodp --version
+    - aodp --help
+
+about:
+  home: 'https://github.com/peterk87/aodp'
+  license: GPL-3.0-only
```
Comment (bgruening) [inline recipes/aodp/meta.yaml:36]: can you please add a license_file

Comment (peterk87) [reply recipes/aodp/meta.yaml:33]: Added in c05b380

File: recipes/aodp/meta.yaml (hunk 1 above)
Comment (peterk87) [reply recipes/aodp/meta.yaml]: Removed in c05b380. Thanks for the feedback @bgruening !

**Comment by peterk87** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**